tests/e2e/uninstall.go

```diff
@@ -0,0 +1,194 @@
+	It("Uninstalling Prometheus-operator Helm chart", func() {
+		uninstallHelmChartIfExists(kubectlOptions, "prometheus-operator", true)
+		By("Verifying Prometheus-operator helm chart resources cleanup")
+
+		k8sResourceKinds, err := listK8sResourceKinds(kubectlOptions, "")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		prometheusAvailableResourceKinds := stringSlicesInstersect(prometheusCRDs(), k8sResourceKinds)
+		prometheusAvailableResourceKinds = append(prometheusAvailableResourceKinds, basicK8sResourceKinds()...)
+
+		remainedResources, err := getK8sResources(kubectlOptions,
+			prometheusAvailableResourceKinds,
+			fmt.Sprintf(managedByHelmLabelTemplate, prometheusOperatorHelmDescriptor.ReleaseName),
+			"",
+			kubectlArgGoTemplateKindNameNamespace,
+			"--all-namespaces")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(remainedResources).Should(BeEmpty())
+	})
+}
+
+// requireRemovePrometheusOperatorCRDs deletes the Prometheus-operator CRDs
+func requireRemovePrometheusOperatorCRDs(kubectlOptions k8s.KubectlOptions) {
+	It("Removing prometheus-operator CRDs", func() {
+		for _, crd := range prometheusCRDs() {
+			deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
+		}
+	})
+}
+
+// requireUninstallingCertManager uninstall Cert-manager Helm chart and
+// remove CRDs.
+func requireUninstallingCertManager(kubectlOptions k8s.KubectlOptions) {
+	When("Uninstalling zookeeper-operator", func() {
+		requireUninstallingCertManagerHelmChart(kubectlOptions)
+		requireRemoveCertManagerCRDs(kubectlOptions)
+	})
+}
+
+// requireUninstallingCertManagerHelmChart uninstalls cert-manager helm chart
+// and checks the success of that operation.
+func requireUninstallingCertManagerHelmChart(kubectlOptions k8s.KubectlOptions) {
+	It("Uninstalling Cert-manager Helm chart", func() {
+		uninstallHelmChartIfExists(kubectlOptions, certManagerHelmDescriptor.ReleaseName, true)
+		By("Verifying Cert-manager helm chart resources cleanup")
+
+		k8sResourceKinds, err := listK8sResourceKinds(kubectlOptions, "")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		certManagerAvailableResourceKinds := stringSlicesInstersect(certManagerCRDs(), k8sResourceKinds)
+		certManagerAvailableResourceKinds = append(certManagerAvailableResourceKinds, basicK8sResourceKinds()...)
+
+		remainedResources, err := getK8sResources(kubectlOptions,
+			certManagerAvailableResourceKinds,
+			fmt.Sprintf(managedByHelmLabelTemplate, certManagerHelmDescriptor.ReleaseName),
+			"",
+			kubectlArgGoTemplateKindNameNamespace,
+			"--all-namespaces")
+		Expect(err).ShouldNot(HaveOccurred())
+
+		Expect(remainedResources).Should(BeEmpty())
+	})
+
+}
+
+// requireRemoveKoperatorCRDs deletes the cert-manager CRDs
+func requireRemoveCertManagerCRDs(kubectlOptions k8s.KubectlOptions) {
+	It("Removing cert-manager CRDs", func() {
+		for _, crd := range certManagerCRDs() {
+			deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
```

Comment:
**Required**: We **MUST** check potential errors here.

```suggestion
			err := deleteK8sResourceNoErrNotFound(kubectlOptions, defaultDeletionTimeout, crdKind, crd)
			
			Expect(err).NotTo(HaveOccurred())
```

Reply:
fixed: [3efc5f9](https://github.com/banzaicloud/koperator/pull/987/commits/3efc5f92e73093a477784196d6bcf64ffcc0a417)